pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -194,36 +206,37 @@ func (p *MellanoxPlugin) Apply() error {
 
 // nicHasExternallyManagedPFs returns true if one of the ports(interface) of the NIC is marked as externally managed
 // in StoreManagerInterface.
-func (p *MellanoxPlugin) nicHasExternallyManagedPFs(nicPortsMap map[string]sriovnetworkv1.InterfaceExt) bool {
+func (p *MellanoxPlugin) nicHasExternallyManagedPFs(nicPortsMap map[string]sriovnetworkv1.InterfaceExt) (bool, error) {
 	for _, iface := range nicPortsMap {
 		pfStatus, exist, err := p.helpers.LoadPfsStatus(iface.PciAddress)
 		if err != nil {
 			log.Log.Error(err, "failed to load PF status from disk", "address", iface.PciAddress)
-			continue
+			return false, err
 		}
 		if !exist {
 			continue
 		}
 		if pfStatus.ExternallyManaged {
 			log.Log.V(2).Info("PF is extenally managed, skip FW TotalVfs reset")
-			return true
+			return true, nil
 		}
 	}
-	return false
+	return false, nil
 }
 
-// nicNotConfiguredByOperator returns true if one of the ports(interface) of the NIC is not configured by operator
-func (p *MellanoxPlugin) nicNotConfiguredByOperator(nicPortsMap map[string]sriovnetworkv1.InterfaceExt) bool {
+// nicConfiguredByOperator returns true if one of the ports(interface) of the NIC is configured by operator
+func (p *MellanoxPlugin) nicConfiguredByOperator(nicPortsMap map[string]sriovnetworkv1.InterfaceExt) (bool, error) {
 	for _, iface := range nicPortsMap {
 		_, exist, err := p.helpers.LoadPfsStatus(iface.PciAddress)
 		if err != nil {
 			log.Log.Error(err, "failed to load PF status from disk", "address", iface.PciAddress)
-			continue
+			return false, err
 		}
 		if exist {
 			log.Log.V(2).Info("PF configured by the operator", "interface", iface)
-			return true
```

Comment:
so this was the bug ?

i see you reversed the meaning of this function

Reply:
done